client/client/app/components/ngbBrowser/ngbBrowser.controller.js

```diff
@@ -36,11 +43,27 @@ export default class ngbBrowserController extends baseController {
         this.isChromosome = chromosome !== null;
         this.isProject = reference !== null && reference !== undefined;
         this.isProjectLoading = this.projectContext.referenceIsPromised;
-
         this.$timeout(::this.$scope.$apply);
     }
 
     browserHomePageUrl() {
-        return this.$sce.trustAsResourceUrl(this.projectContext.browserHomePageUrl);
+        return this.$sce.trustAsResourceUrl(this.homeUrl);
+    }
+
+    homeUrlIsMd() {
+        return this.homeUrl && /.md$/.test(this.homeUrl);
+    }
+    async getMarkdown() {
+        if (this.homeUrl) {
+            try {
+                this.pageIsLoading = true;
+                this.markdown = await this.ngbBrowserService.getMarkdown(this.homeUrl);
+            } catch(err) {
+                this.errorMessages.push(err.message);
+            } finally {
+                this.pageIsLoading = false;
+                this.$scope.$apply();
+            }
+        }
```

Comment:
`this.errorMessages` will never be reset; consider the following scenario:
- we have `this.homeUrl` set to invalid resource;
- we receive *fetch* error => we display this error using `this.errorMessages`;
- we receive new `this.homeUrl` (this is quite rare case in real environment, but still - we are listening to changes on `this.homeUrl`);
- we re-fetch `this.homeUrl` again, without any errors (but `this.errorMessages` still contains previous error)
- we display both content and previous error